src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -7656,15 +7656,22 @@ public Encoding Encoding
 
             set
             {
-                // Check for UTF-7 by checking for code page 65000
-                // See: https://docs.microsoft.com/en-us/dotnet/core/compatibility/corefx#utf-7-code-paths-are-obsolete
-                if (value != null && value.CodePage == 65000)
-                {
-                    _provider.WriteWarning(PathUtilsStrings.Utf7EncodingObsolete);
-                }
+                WarnIfObsolete(value);
```

Comment:
What about rename to `WarnIfUtf8()`?

Reply:
I used an existing method name from `EncodingConversion`.

`WarnIfUtf7` makes more sense, but I'd rather keep the same name.

Reply:
I'd prefer more informative name in both cases.